web/modules/sheaf/src/components/Wikitext.svelte

```diff
@@ -70,53 +79,54 @@
   }
 
   const render = createMutatingLock(async (wikitext: WikitextInput) => {
-    const displayIndicatorTimeout = setTimeout(() => (rendering = true), 100)
+    const displayIndicatorTimeout = setTimeout(() => (rendering = true), 500)
     const measure = perfy()
+
     if (typeof wikitext === "function") wikitext = wikitext()
     wikitext = await wikitext
-    const result: Rendered = wikitext
-      ? typeof wikitext !== "string"
-        ? wikitext
-        : await FTML.renderHTML(wikitext)
-      : { html: "", styles: [""] }
-    perfRender = measure()
-    clearTimeout(displayIndicatorTimeout)
-    return result
-  })
 
-  // TODO: Security audit of this - how much should we trust FTML output right now?
+    return await idleCallback(async () => {
+      const result: Rendered = wikitext
+        ? typeof wikitext !== "string"
+          ? wikitext
+          : await FTML.renderHTML(wikitext)
+        : { html: "", styles: [""] }
+      perfRender = measure()
+      clearTimeout(displayIndicatorTimeout)
+      return result
+    })
+  })
 
-  const update = createAnimQueued(async ({ html, styles }: Rendered) => {
+  const update = createMutatingLock(async ({ html, styles }: Rendered) => {
     if (!element) return
 
-    // there are better ways to do this, but this is mostly just a development tool
-    // this prevents the console from getting spammed with
-    // crossorigin or missing link errors
-    if (offline) {
-      html = html.replaceAll(
-        /<(img|iframe)[^]+?>/g,
-        "<div>Offline Replacement Element</div>"
-      )
-    }
+    const fragment = await idleCallback(() => toFragment(html))
+
+    await animationFrame(() => {
+      if (morph) {
+        const oldBody = element.querySelector("wj-body")
+        const newBody = fragment.querySelector("wj-body")
 
-    const fragment = toFragment(html)
-    if (morph) {
-      morphdom(element, fragment, {
-        childrenOnly: true,
-        onBeforeElUpdated: function (fromEl, toEl) {
-          if (fromEl.isEqualNode(toEl)) return false
-          return true
+        if (!newBody || !oldBody) {
+          element.innerText = ""
+          element.appendChild(fragment)
+          return
         }
-      })
-    } else {
-      element.innerText = ""
-      element.append(fragment)
-    }
 
-    // prepend style with a index comment so that each style string is unique
-    stylesheets = styles.map(
-      (style, idx) => `\n/* stylesheet ${idx + 1} */\n\n${style}\n`
-    )
+        micromorph(oldBody, newBody)
+      } else {
+        element.innerText = ""
+        element.appendChild(fragment)
+      }
+    })
+
+    await animationFrame(() => {
+      // prepend style with a index comment so that each style string is unique
+      stylesheets = styles.map(
+        (style, idx) => `\n/* stylesheet ${idx + 1} */\n\n${style}\n`
+      )
```

Comment:
Interesting, why is this necessary?

Reply:
The stylesheets are uniquely identified by their contents. What is uniquely identifying them is Svelte - below in the markup there is a `{#each}` block that is putting the stylesheet strings into the `<head>`. When the stylesheets array updates, it compares the previous list to the new one.

If you had two stylesheets with the same contents, it would confuse Svelte.